# Task 3
A'zra Zulaikha Binti Zainol
P137262

Find any dataset from any website (eg. kaggle, open data etc.). Your data must have at least 500 instances and seven numerical variables. Your variables can be both numerical and categorical, but must have at least seven numerical variables. Using pandas library, perform the following instruction:

- a) Find the heading. What can you explain?
- b) Is there any missing values?
- c) Sort the columns according to the ascending alphabetical order.
- d) Compute descriptive statistics such as maximum, minimum, mean, median, standard deviation, skewness and kurtosis for each of these numerical variables. Give explanation on these findings.

Based on the above, write an essay which include the following:
1. Introduction and objectives.
2. Explanation on the data (Briefly explain your dataset, including what are the variables involved)
3. Discussion on the finding based on your objectives and your findings from part (d) above.
4. Conclusion
Your essay must be at least 1000 words.

In [1]:
import pandas as pd

In [2]:
coff = pd.read_csv("Coffee_Chain_Sales.csv")

In [3]:
print(coff.head())
# retrieve a preview of the dataset

   Area Code  Cogs  DifferenceBetweenActualandTargetProfit       Date  \
0        303    51                                     -35  10/1/2012   
1        970    52                                     -24  10/1/2012   
2        409    43                                     -22  10/2/2012   
3        850    38                                     -15  10/3/2012   
4        562    72                                       6  10/4/2012   

   Inventory Margin  Margin   Market_size   Market  Marketing Product_line  \
0               503      71  Major Market  Central         46       Leaves   
1               405      71  Major Market  Central         17       Leaves   
2               419      64  Major Market    South         13       Leaves   
3               871      56  Major Market     East         10       Leaves   
4               650     110  Major Market     West         23       Leaves   

   ...     Product Profit  Sales       State Target_cogs  Target_margin  \
0  ...       Lemo

In [4]:
# changing the date variable in the data frame to proper date format for proper analysis
coff['Date'] = pd.to_datetime(coff['Date'])

## a) Find the heading

In [5]:
print(coff.info())
#retrieves the information about the dataset

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1062 entries, 0 to 1061
Data columns (total 21 columns):
 #   Column                                  Non-Null Count  Dtype         
---  ------                                  --------------  -----         
 0   Area Code                               1062 non-null   int64         
 1   Cogs                                    1062 non-null   int64         
 2   DifferenceBetweenActualandTargetProfit  1062 non-null   int64         
 3   Date                                    1062 non-null   datetime64[ns]
 4   Inventory Margin                        1062 non-null   int64         
 5   Margin                                  1062 non-null   int64         
 6   Market_size                             1062 non-null   object        
 7   Market                                  1062 non-null   object        
 8   Marketing                               1062 non-null   int64         
 9   Product_line                            1062 non-nul

In [6]:
# retrieve the column names
print(coff.columns)
print("\n-----------------------------------------------------------------------")
print(len(coff.columns))
print(len(coff))

Index(['Area Code', 'Cogs', 'DifferenceBetweenActualandTargetProfit', 'Date',
       'Inventory Margin', 'Margin', 'Market_size', 'Market', 'Marketing',
       'Product_line', 'Product_type', 'Product', 'Profit', 'Sales', 'State',
       'Target_cogs', 'Target_margin', 'Target_profit', 'Target_sales ',
       'Total_expenses', 'Type'],
      dtype='object')

-----------------------------------------------------------------------
21
1062


### Explain the headings:

The headings indicate that this data set is about the sales and profit of coffee chains based on the area of the coffee chain and the type of product. It includes details like marketing and inventory costs, market size and sales for specific product types. There are 21 variables in the data set and 1062 rows.

### The explanation of each heading is as below:

Area code: This is the code of the place where the coffee chain operates.

Cogs: This is the 'Cost of Goods Sold', which is the cost of producing or making the products that are sold.

DifferenceBetweenActualandTargetProfit: This is the 'Actual Profit' - 'Target Profit'. 'Actual Profit' is the profit amount and 'Target Profit' is the profit goals. Positive values mean that the goals are reached and vice versa.

Date: This is the date of the sales.

Inventory Margin: This is the 'Revenue from Selling Inventory' - 'Cost of Maintaining Inventory'. 'Revenue from Selling Inventory' is the returns from selling the inventory and 'Cost of Maintaining Inventory' is the cost to store the inventory.

Margin: This is the profit percentage earned from sales.

Market_size: This is the market size, which means the possible customer base, in the area of the coffee chains.

Market: This is the location of the coffee chain in the area.

Marketing: This is the cost of marketing (eg: advertising).

Product_line: This is the main category of products which can be 'Leaves' for tea or 'Beans' for coffee.

Product_type: This is the type of tea and coffee.

Product: This is the sub categories of tea and coffee types.

Profit: This is the profit made by the coffee chain.

Sales: This is the total revenue earned by the coffee chain.

State: This is the state where the coffee chain is located.

Target_cogs: This is the target for Cost of Goods Sold.

Target_margin: This is the target margin.

Target_profit: This is the target profit.

Target_sales: This is the target sales.

Total_expenses: This is the total expenses spent for running the coffee chain.

Type: This is the type of caffiene which can be 'Decaff' or 'Regular'.

## b) Is there any missing values?

In [7]:
# check missing values
print(coff.isnull().sum())
print("________________________________________________________")
print("\nThere are no missing values in the data set.")

Area Code                                 0
Cogs                                      0
DifferenceBetweenActualandTargetProfit    0
Date                                      0
Inventory Margin                          0
Margin                                    0
Market_size                               0
Market                                    0
Marketing                                 0
Product_line                              0
Product_type                              0
Product                                   0
Profit                                    0
Sales                                     0
State                                     0
Target_cogs                               0
Target_margin                             0
Target_profit                             0
Target_sales                              0
Total_expenses                            0
Type                                      0
dtype: int64
________________________________________________________

There

## c) Sort the columns according to the ascending alphabetical order.

In [8]:
# sorting the columns into ascending alphabetical order based on column name
print(coff.sort_index(axis = 1))

      Area Code  Cogs       Date  DifferenceBetweenActualandTargetProfit  \
0           303    51 2012-10-01                                     -35   
1           970    52 2012-10-01                                     -24   
2           409    43 2012-10-02                                     -22   
3           850    38 2012-10-03                                     -15   
4           562    72 2012-10-04                                       6   
...         ...   ...        ...                                     ...   
1057        775   250 2015-08-23                                     133   
1058        971    88 2015-08-24                                      48   
1059        775   294 2015-08-25                                    -285   
1060        503   134 2015-08-26                                      80   
1061        435    20 2015-08-27                                     -22   

      Inventory Margin  Margin   Market   Market_size  Marketing     Product  \
0      

## d) Compute descriptive statistics
- maximum
- minimum
- mean
- median
- standard deviation
- skewness
- kurtosis

In [9]:
# retrieve a summary of the data which shows the number of values, mean, minimum, 1st quartile, median, 3rd quartile, maximum, and standard deviation
print(coff.describe())

         Area Code         Cogs  DifferenceBetweenActualandTargetProfit  \
count  1062.000000  1062.000000                             1062.000000   
mean    587.030132    82.399247                                0.387006   
min     203.000000     0.000000                             -369.000000   
25%     425.000000    41.000000                              -15.000000   
50%     573.000000    57.000000                               -3.000000   
75%     774.000000   101.000000                               13.000000   
max     985.000000   294.000000                              249.000000   
std     225.299162    64.824295                               44.331180   

                                Date  Inventory Margin       Margin  \
count                           1062       1062.000000  1062.000000   
mean   2014-03-14 12:01:21.355932416        815.175141   102.423729   
min              2012-10-01 00:00:00      -3534.000000  -294.000000   
25%              2013-06-22 06:00:00    

In [10]:
# retrieve the skewness and kurtosis of the data distribution
coff_num_range = list(range(1, 3)) + list(range(4, 6)) + list(range(8, 9)) + list(range(12, 14)) + list(range(15, 20))
coff_num = coff.iloc[:, coff_num_range]
print("Skewness results:")
print(coff_num.skew())
print("\nKurtosis results:")
print(coff_num.kurtosis())

Skewness results:
Cogs                                      1.536993
DifferenceBetweenActualandTargetProfit   -0.156458
Inventory Margin                          2.604671
Margin                                    1.222219
Marketing                                 1.487507
Profit                                    0.282312
Sales                                     1.756490
Target_cogs                               1.896974
Target_margin                             1.887134
Target_profit                             1.083739
Target_sales                              2.177533
Total_expenses                            1.188945
dtype: float64

Kurtosis results:
Cogs                                       1.609019
DifferenceBetweenActualandTargetProfit    11.518237
Inventory Margin                          21.745677
Margin                                     4.295716
Marketing                                  1.637996
Profit                                     8.973741
Sales                   

Based on the above, write an essay which include the following:
1. Introduction and objectives.
2. Explanation on the data (Briefly explain your dataset, including what are the variables
involved)
3. Discussion on the finding based on your objectives and your findings from part (d)
above.
4. Conclusion

Your essay must be at least 1000 words.

A dataset from kaggle was employed for this task. The dataset contains information between 2012-10-01 and 2015-08-27 about a coffee chain company. Based on the dataset, the company wants to carry out a simple analysis regarding their sales and profit for all of their available branches within the given period. They want to see how much they are earning and also how much they are spending to keep the coffee chains running smoothly and attracting customers.

The dataset has the variable names Area Code, Cogs, DifferenceBetweenActualandTargetProfit, Date, Inventory Margin, Margin, Market_size, Market, Marketing, Product_line, Product_type, Product, Profit, Sales, State, Target_cogs, Target_margin, Target_profit, Target_sales, Total_expenses, and Type.

The numerical variables in the dataset that the company is interested in examining are Cogs, DifferenceBetweenActualandTargetProfit, Margin, Profit, Sales, Target_cogs, Target_margin, Target_profit, Target_sales, and Total_expenses. For these variables, the Cogs variable shows the cost of goods sold, the DifferenceBetweenActualandTargetProfit shows the actual profit subtracted by the target profit, the Margin is the percentage of profit earned from sales, the Profit is the profit earned by the coffee chain, the Sales shows the revenue earned, the Target_cogs shows the target for Cost of Goods Sold, the Target_margin is the target margin, the Target_profit shows the target profit, the Target_sales is the target sales and finally the Total_expenses shows the total expenses spent for running the coffee chain.

Firstly, the company checks sales for the coffee chains based on the Sales column and compares it with the target sales in the Target_sales column. The minimum sales is 21 and the maximum sales is 815 while the minimum target is 0.0 and the maximum target is 960.0. The median sales is 133 and the median target is 120. This may indicate that most coffee chains earned 133 in sales which is similar to the median target. However, based on common knowledge, the minimum sales is pretty low despite no coffee chain getting lower than the minimum target. This may indicate that there are coffee chains that experienced loss. The maximum sales is also very high. The skewness value for sales, 1.7564, indicates that quite a few coffee chains earned high sales meanwhile the rest of the coffee chains still earned similar values to the average sales which is 191 with a standard deviation of 148.27. Compared to the average target, 168.4934, the coffee chains did pretty well on average. The kurtosis is 2.7672 which indicates that there are extreme values which means that some coffee chains have a sales value that is very different from the majority of the coffee chains and there may be some coffee chains earning less or more than average.

The company now wants to identify whether there are losses experienced by their coffee chains. To check, the company refers to the Profit column to see if there are any coffee chains experiencing loss. Based on the output, the minimum value is -605 and maximum value is 646. The negative value shows that there are coffee chains that experienced loss. In comparison to the Target_profit column, the minimum and maximum values are -320 and 470 respectively. The range of profit is wider than the range of target profit. The median of profit is 39.5 which means most of the coffee chains earned profit similar to the median of the target profit which is 40.0. The distribution of the profit has a skewness of 0.28231, a kurtosis of 8.973741, and a mean of 60.5565. The skewness indicates that there is generally a normal distribution but the kurtosis shows there are values that are extreme and vary greatly from the mean with a standard deviation of 100.5166.

The company also wants to check what factors may possibly be affecting the coffee chain with the highest and lowest profits. It was detected that the coffee chain with the lowest profit was located in an area with a small market size and only earned 33 in sales despite spending 111 on marketing. The margin, -294, also indicates that the profit percentage earned from their sales are extremely low. Next, the coffee chain with the highest profit had a major market size and a margin of 526. This means that they earned a large profit from the sales they made. They also spent only 66 on marketing which saved them cost. The company decides to look into this problem further in the future.

After checking the profit and sales for the coffee branches, the overall performance of the company should be evaluated. The company decides to examine the DifferenceBetweenActualandTargetProfit column. On average, the company barely passed their target profit by an average of 0.387 with a standard deviation of 44.3311. The median is -3.0 which shows that the majority of the coffee chains did not pass their profit. The minimum and maximum values are -369 and 249 respectively. The distribution is skewed to the left and has leptokurtic distribution showing that the coffee chains dominantly did not pass their target profit and there are extreme values. 

Now the company wants to check the cost of goods sold (Cogs) against the target cost of goods sold. To do so, the company compares the Cogs and Target_cogs columns respectively. It seems like both the minimum for Cogs and Target_cogs are 0.00. The maximum for Cogs is 294 meanwhile the maximum Target_cogs is 380. The median Cogs is 57.0 and the median Target_cogs is 50.0. The average Cogs is 82.4 while the average Target_cogs is 71.68. This means that on average, the coffee branches passed their target cost of goods sold. Cogs has a skewness of 1.5370 and a kurtosis of 1.6090 showing that there are coffee chains that have higher values of Cogs compared to the average.

Next, the company wants to know the total expenses the coffee chains spent using the Total_expenses column. The minimum and maximum expenses are 11 and 156 respectively. The median is 46 and the average is 53.84 with a standard deviation of 31.70. This indicates that most coffee chains spent around the average value to keep the coffee chain running. The distribution is skewed to the right by 2.1775 and has a kurtosis of 1.0635 showing that there are a few coffee chains that are spending more than the average.

Finally, the company examines the Margin and Target_margin columns to compare the percentage of profit and the target profit percentage. The minimum and maximum for the Margin is -294 and 526 meanwhile the minimum and maximum for Target_margin is -210 and 580. The median margin is 73 and the median target margin is 70. The mean margin is 102.42 and the mean target margin is 96.82. The median and mean indicates that most coffee chains passed the target margin. Both the margin and the target margin have a leptokurtic distribution and are skewed to the right. This means that there are extreme values towards the right side of the graph indicating that there are coffee chains with high margin and target margin values.

To conclude, the company did well where they mostly reached their targets in most areas. Regardless, they still have some coffee chains that need proper management so that they do not overspend on marketing but not gain sales. The company may do further analysis to identify which coffee chains need more management to avoid making the wrong decisions or to move the coffee chains location to a more populated area with larger market size.

In [11]:
## EXTRA CODES FOR ANALYSIS:

In [12]:
type(coff)

pandas.core.frame.DataFrame

In [13]:
print(coff['Date'].min())
print(coff['Date'].max())

2012-10-01 00:00:00
2015-08-27 00:00:00


In [14]:
# row for lowest profit
coff[coff["Profit"] == -605].index[0]

1059

In [15]:
# row for highest profit
coff[coff["Profit"] == 646].index[0]

690

In [16]:
# lowest profit
coff.iloc[1059, ]

Area Code                                                 775
Cogs                                                      294
DifferenceBetweenActualandTargetProfit                   -285
Date                                      2015-08-25 00:00:00
Inventory Margin                                         8252
Margin                                                   -294
Market_size                                      Small Market
Market                                                   West
Marketing                                                 111
Product_line                                           Leaves
Product_type                                              Tea
Product                                             Green Tea
Profit                                                   -605
Sales                                                      33
State                                                  Nevada
Target_cogs                                               210
Target_m

In [17]:
# highest profit
coff.iloc[690, ]

Area Code                                                 212
Cogs                                                      239
DifferenceBetweenActualandTargetProfit                    196
Date                                      2014-08-21 00:00:00
Inventory Margin                                         1197
Margin                                                    526
Market_size                                      Major Market
Market                                                   East
Marketing                                                  66
Product_line                                            Beans
Product_type                                         Espresso
Product                                      Regular Espresso
Profit                                                    646
Sales                                                     815
State                                                New York
Target_cogs                                               210
Target_m